web/html/src/components/table/Table.test.js

```diff
@@ -117,4 +117,28 @@ describe("Table component", () => {
       })
     ).not.toBe(null);
   });
+
+  test("loading indicator for tables using SimpleDataProvider", async () => {
+    const data = [{value: "Value 0"}];
+
+    const { rerender } = render(
+      <Table {...baseProps} loading={true} />
+    );
+
+    // Check if loading indicator appears
+    expect(screen.queryByText(baseProps.loadingText)).not.toBe(null);
+
+    rerender(
+      <Table {...baseProps} data={data} loading={false} >
+        <Column columnKey="value" cell={item => item.value} />
+      </Table>
+    );
+    await waitForElementToBeRemoved(() =>
+      screen.queryByText(baseProps.loadingText)
+    );
+
+    // Check if loading indicator disappears and data is displayed
+    expect(screen.queryByText(baseProps.loadingText)).toBe(null);
```

Comment:
Realistically this is redundant with the `waitForElementToBeRemoved`, but I think it's completely fine to be verbose here with the intention. Looks good to me. 👍

In the long run, I would hope a smoother abstraction around loading will emerge as we write more tests for the table, but time will tell.